### PR TITLE
Support multiple repos and add-ons through cmd args

### PR DESCRIPTION
**Pull request by slgobinath**
@Razzeee 
This PR supports multiple add-ons and/or repositories to be passed as command line argument. If none is passed, treat current directory as a repo.

For repositories their respective `.tests-config.json` is used. For add-ons, if there is a `tests-config.json` in the current directory, that will be used.

Tested using two repos (krypton and jarvis) using different `.tests-config.json` settings.

Please note that the command line argument is renamed from `add-on` to `dir`. If you have a better name, please let me know.

Thanks

### CODE REVIEW

File: kodi_addon_checker/__main__.py [hunk 1]
```diff
@@ -39,15 +40,47 @@ def main():
...
-                        help="optional add-on directories")
+    parser.add_argument("dir", type=dir_type, nargs="*", help="optional add-on or repo directories")
     ConfigManager.fill_cmd_args(parser)
     args = parser.parse_args()
 
-    repo_path = os.path.abspath(os.getcwd())
-    config = Config(repo_path, args)
-    ConfigManager.process_config(config)
-    check_repo(config, repo_path, args.add_on)
+    current_dir = os.path.abspath(os.getcwd())
```
Comment (Razzeee) [inline kodi_addon_checker/__main__.py]: If we pass the directory to the checker, this folder might be completly unrelated. Therefore I think we should never use this in that case.

File: kodi_addon_checker/__main__.py [hunk 2]
```diff
@@ -39,15 +40,47 @@ def main():
...
+    if args.dir:
+        report = Report(current_dir)
+        for directory in args.dir:
+            if os.path.isfile(os.path.join(directory, "addon.xml")):
+                report.add(Record(INFORMATION, "Checking add-on %s" % directory))
+                # For add-ons try to load .tests-config.json from current directory
+                config = Config(current_dir, args)
```
Comment (Razzeee) [inline kodi_addon_checker/__main__.py]: Go for `os.path.abspath(directory)` here. That way developers can just have a local gitignored copy of the .test-config.json they want to use int the addons root dir.

File: kodi_addon_checker/__main__.py [hunk 3]
```diff
@@ -39,15 +40,47 @@ def main():
...
+                addon_report = check_addon.start(os.path.abspath(directory), config)
+                report.add(addon_report)
+            else:
+                repo_path = os.path.abspath(directory)
```
Comment (Razzeee) [inline kodi_addon_checker/__main__.py]: Can we print something for this case too? Just to communicate that we're using repo handling?

File: kodi_addon_checker/__main__.py [hunk 4]
```diff
@@ -39,15 +40,47 @@ def main():
...
+                config = Config(repo_path, args)
+                repo_report = check_repo(repo_path, config)
+                report.add(repo_report)
+    else:
+        if os.path.isfile(os.path.join(current_dir, "addon.xml")):
+            # Current directory is an add-on
+            report = Report(current_dir)
+            report.add(Record(INFORMATION, "Checking add-on %s" % current_dir))
+            config = Config(None, args)
+            addon_report = check_addon.start(os.path.abspath(current_dir), config)
```
[the remark this answers is not in the document]
Comment (slgobinath) [reply kodi_addon_checker/__main__.py]: I have moved `Checking ...` INFO to relevant checkers. Now check_repo adds `Checking repository ...` and check_addon.start adds `Checking add-on ...` which avoids code duplication and improve cohesion.

**Comment by Rechi** [comment]
@slgobinath please remove the merge commit.
If you want the latest changes of the base branch in your feature branch rebase on it instead of merge it back.

**Comment by slgobinath** [comment]
@Rechi and @Razzeee 
Please note that I am closing this PR in favor of https://github.com/xbmc/addon-check/pull/67 with clean commits history.

**Pull request closed without merging**